proxy_server.py

```diff
@@ -59,12 +61,16 @@ class AddProxyServerHandler(webapp2.RequestHandler):
   """Handler for adding new proxy servers."""
 
   @admin.RequireAdmin
+  @OAUTH_DECORATOR.oauth_required
+  @dasher_admin.DasherAdminAuthRequired
```

Comment:
I think we should combine the 2 different admin decorators into one.  For example, the @admin.RequireAdmin can be made to look like below, and we can add a IsDomainAdmin(user) method in google_directory_service.  This will make the code much cleaner.

Also, should change "dasher" to "domain" so that it's easier for external users to understand.

    if not user:
      logging.error('user is not logged in')
      self.abort(403)
    if not users.is_current_user_admin():
      logging.error('user is not an admin')
      self.abort(403)
   if not directory_service.IsDomainAdmin(user):
      logging.error('user is not an domain admin')
      self.abort(403)